### PR TITLE
Return `invalid_search_facets` rather than `bad_request` when using facet on a non filterable attribute

### PR DESCRIPTION
**Pull request by dureuill**
# Pull Request

## Related issue

Fixes https://github.com/meilisearch/meilisearch/issues/3384

## What does the PR does

- title
- also adds a test

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch/tests/search/errors.rs [hunk 1]
```diff
@@ -321,6 +321,25 @@ async fn search_bad_facets() {
...
+    snapshot!(code, @"400 Bad Request");
+    snapshot!(json_string!(response), @r###"
+    {
+      "message": "Invalid facet distribution, the fields `doggo` are not set as filterable.",
+      "code": "invalid_search_facets",
+      "type": "invalid_request",
+      "link": "https://docs.meilisearch.com/errors#invalid-search-facets"
+    }
+    "###);
+    // Can't make the `attributes_to_highlight` fail with a get search since it'll accept anything as an array of strings.
```
Comment (irevoire) [inline meilisearch/tests/search/errors.rs:340]: Could you write a test with a `search_get` since you actually can in this case

**Review by irevoire — CHANGES_REQUESTED** [review]
Nice, i would just like to slightly improve the test

File: meilisearch/tests/search/errors.rs (hunk 1 above)
Comment (dureuill) [reply meilisearch/tests/search/errors.rs]: done

**Review by irevoire — APPROVED** [review]
Nice, thank you :ok: 
bors merge

**Pull request merged**